File: java/703-Kth-Largest-Element-in-a-Stream.java
class KthLargest {

    final PriorityQueue<Integer> heap = new PriorityQueue<>();
    final int k;

    public KthLargest(int k, int[] nums) {
        this.k = k;
        for (int n : nums) add(n);
    }

    public int add(int val) {
        if (heap.size() < k) heap.offer(val); //for adding the values of the array
        else if (val > heap.peek()) {
            heap.poll(); //remove the top element
            heap.add(val); //add the new element
        }
        return heap.peek();
    }
}

File: kotlin/703-kth-largest-element-in-a-stream.kt
class KthLargest(k: Int, nums: IntArray) {
    val minHeap = PriorityQueue<Int>{ a: Int, b: Int ->
        a - b
    }
    var k = Integer.MIN_VALUE
    init{
        this.k = k
        for(num in nums)
            minHeap.add(num)
        while(minHeap.size > k)
            minHeap.poll()
    }
    fun add(`val`: Int): Int {
        minHeap.add(`val`)
        if(minHeap.size > k)
            minHeap.poll()
        return minHeap.peek()
    }

}
